components/Feature/Services/index.js

```diff
@@ -1,48 +1,50 @@
 import { useState } from 'react';
 import ResourceCard from 'components/Feature/VulnerabilitiesGrid/ResourceCard';
-import {
-  Accordion,
-  AccordionItem
-} from 'components/Form';
+import { Accordion, AccordionItem } from 'components/Form';
 
 const Services = ({ resources, taxonomies }) => {
   const [expandedGroups, setExpandedGroups] = useState({});
 
+  taxonomies.forEach(
+    taxonomy =>
+      (taxonomy.resources = resources
```

Comment:
I don't think you should mutate the object that you've received through the props here.

Is there a clearer way you can write this?

Reply:
Yeah, I guess I could copy it Into another object 🤷

Reply:
For example, you could have `taxonomiesToRender` which does something like

```
const taxonomiesToRender = taxonomies.filter(
  taxonomy => resources.some(
    resource => resource.categoryName === taxonomy.name
  )
)
```

Reply:
Oooh, I like it a lot. I forget that world outside of map exists 😂 I'll add that now

Reply:
Hmm, now to think of it, this forces to have ``` resource => resource.categoryName === taxonomy.name``` check twice. Once here and also once as we render the resources. I'm tempted to put it into one object before the render, e.g. like this:
```
  const taxonomiesToRender = taxonomies.filter(taxonomy =>
    resources.some(resource => resource.categoryName === taxonomy.name)
  );

  taxonomiesToRender.forEach(
    taxonomy =>
      (taxonomy.resources = resources
        .map(resource => {
          if (taxonomy.name == resource.categoryName) return resource;
        })
        .filter(x => x))
  );
  ```
But that almost seems like the first part of it is irrelevant. Any ideas?